sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -61,7 +61,13 @@ object JDBCRDD extends Logging {
       try {
         val rs = statement.executeQuery()
         try {
-          JdbcUtils.getSchema(rs, dialect)
+          val metaStructType = JdbcUtils.getSchema(rs, dialect)
```

Comment:
Let us add a new boolean parameter `alwaysNullable`  to `getSchema`

Reply:
Isn't that bad to hide useful schema info on `getSchema()` level? 
What if nullable=false is usable for other purposes, like query optimization?